LinearRegression/LinearRegressionBackend/MLNeuralNetwork/Layer.cs

```diff
@@ -25,18 +30,29 @@ public Layer(Matrix<double> weights, Vector<double> biases, IActivationFunction
             ActivationFunction = activationFunction;
         }
 
-        public Vector<double> WeightedSums(Vector<double> previousLayer)
+        public void PropagateForward(Vector<double> inputData)
+        {
+            WeightedSums = CalculateWeightedSums(inputData);
+            Activations = CalculateActivations(WeightedSums);
+        }
+
+        private Vector<double> CalculateWeightedSums(Vector<double> inputData)
         {
-            Debug.Assert(previousLayer.Count == Weights.ColumnCount);
-            return Weights * previousLayer + Biases;
+            Debug.Assert(inputData.Count == Weights.ColumnCount);
+            return Weights * inputData + Biases;
         }
 
-        public Vector<double> Activations(Vector<double> weightedSums)
+        private Vector<double> CalculateActivations(Vector<double> weightedSums)
         {
             return Vector<double>.Build.Dense(
                 weightedSums.Count,
                 i => ActivationFunction.Activation(weightedSums[i])
             );
         }
+
+        public void Update(double rateOfChange, double learningRate)
+        {
+            Weights += learningRate * rateOfChange * WeightsGradient;
```

Comment:
The correct formula would be:
weights = weights - alpha*weightsGradient

Same for the bias:
bias = bias - alpha*biasGradient

Reply:
Yeah, I messed up the sign. I included the `rateOfChange` to make the step in the negative gradient direction proportional to the partial derivative at that point. I saw this in the 3blue1brown videos and thought we could use it.